[ONNX] Set correct cuda.current_device for multi-device onnx performance bench

BowenBao opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #115773
* #115673
* __->__ #115670

Otherwise `torch.cuda.synchronize()` works on a different device from the one that
runs PyTorch model, which lead to incorrect performance number.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @aakhundov @kadeng

BowenBao committed:
[ONNX] Set correct cuda.current_device for multi-device onnx performance bench

Otherwise `torch.cuda.synchronize()` works on a different device from the one that
runs PyTorch model, which lead to incorrect performance number.

[ghstack-poisoned]

thiagocrepaldi reviewed (approved):
LGTM. 

I'd improve the comment to explain why device `0` is the correct device on the first run and why device `1` is the correct device for the second.

BowenBao committed:
add comment on "[ONNX] Set correct cuda.current_device for multi-device onnx performance bench"


Otherwise `torch.cuda.synchronize()` works on a different device from the one that
runs PyTorch model, which lead to incorrect performance number.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

[ghstack-poisoned]

thiagocrepaldi commented on benchmarks/dynamo/common.py:
```diff
@@ -791,6 +791,8 @@ def timed_onnx(model, onnx_model: OnnxModel, inputs):
         )
         if torch.cuda.device_count() > 1:
             # Manually set correct torch.cuda.current_device to ensure torch.cuda.synchronize() works as intended.
+            # When there are more than 1 cuda devices, the first one is used for pytorch eager.
```
perfect!
